chain/x/oracle/module.go

```diff
@@ -107,3 +115,156 @@ func (am AppModule) ExportGenesis(ctx sdk.Context, cdc codec.JSONMarshaler) json
+	counterpartyVersion string,
+) error {
+	// TODO: Enforce ordering, currently relayers use ORDERED channels
+
+	// if version != types.Version {
+	// 	return sdkerrors.Wrapf(porttypes.ErrInvalidPort, "invalid version: %s, expected %s", version, "ics20-1")
+	// }
+
+	// if counterpartyVersion != types.Version {
+	// 	return sdkerrors.Wrapf(porttypes.ErrInvalidPort, "invalid counterparty version: %s, expected %s", counterpartyVersion, "ics20-1")
+	// }
+
+	// Claim channel capability passed back by IBC module
+	if err := am.keeper.ScopedKeeper.ClaimCapability(ctx, chanCap, ibctypes.ChannelCapabilityPath(portID, channelID)); err != nil {
+		return sdkerrors.Wrap(channel.ErrChannelCapabilityNotFound, err.Error())
+	}
+
+	return nil
+}
+
+func (am AppModule) OnChanOpenAck(
+	ctx sdk.Context,
+	portID,
+	channelID string,
+	counterpartyVersion string,
+) error {
+	// if counterpartyVersion != types.Version {
+	// 	return sdkerrors.Wrapf(porttypes.ErrInvalidPort, "invalid counterparty version: %s, expected %s", counterpartyVersion, "ics20-1")
+	// }
+	return nil
+}
+
+func (am AppModule) OnChanOpenConfirm(
+	ctx sdk.Context,
+	portID,
+	channelID string,
+) error {
+	return nil
+}
+
+func (am AppModule) OnChanCloseInit(
+	ctx sdk.Context,
+	portID,
+	channelID string,
+) error {
+	// Disallow user-initiated channel closing for transfer channels
+	return sdkerrors.Wrap(sdkerrors.ErrInvalidRequest, "user cannot close channel")
+}
+
+func (am AppModule) OnChanCloseConfirm(
+	ctx sdk.Context,
+	portID,
+	channelID string,
+) error {
+	return nil
+}
+
+func (am AppModule) OnRecvPacket(
+	ctx sdk.Context,
+	packet channeltypes.Packet,
+) (*sdk.Result, error) {
+	var requestData OracleRequestPacketData
+	if err := types.ModuleCdc.UnmarshalJSON(packet.GetData(), &requestData); err == nil {
+		calldata, err := hex.DecodeString(requestData.Calldata)
+		if err != nil {
+			return nil, err
+		}
+		newMsg := NewMsgRequestData(
+			requestData.OracleScriptID, calldata, requestData.AskCount,
+			requestData.MinCount, requestData.ClientID,
+			sdk.AccAddress([]byte("Unknown")),
+		)
+		return handleMsgRequestDataIBC(
+			ctx, am.keeper, newMsg, packet.GetDestPort(), packet.GetDestChannel(),
+		)
+	}
+	// acknowledgement := FungibleTokenPacketAcknowledgement{
+	// 	Success: true,
+	// 	Error:   "",
+	// }
+	// if err := am.keeper.OnRecvPacket(ctx, packet, data); err != nil {
+	// 	acknowledgement = FungibleTokenPacketAcknowledgement{
+	// 		Success: false,
+	// 		Error:   err.Error(),
+	// 	}
+	// }
+
+	// if err := am.keeper.ChannelKeeper.PacketExecuted(ctx, ,packet, acknowledgement.GetBytes()); err != nil {
+	// 	return nil, err
+	// }
+
+	// ctx.EventManager().EmitEvent(
+	// 	sdk.NewEvent(
+	// 		types.EventTypePacket,
+	// 		sdk.NewAttribute(sdk.AttributeKeyModule, types.AttributeValueCategory),
+	// 		// sdk.NewAttribute(types.AttributeKeyReceiver, data.Receiver),
+	// 		// sdk.NewAttribute(types.AttributeKeyValue, data.Amount.String()),
+	// 	),
+	// )
```

Comment:
We can remove these code for now because we don't need to send ackPacket to counter party